src/foam/dao/FixedSizeDAO.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * @license
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+foam.CLASS({
+  package: 'foam.dao',
+  name: 'FixedSizeDAO',
+  extends: 'foam.dao.ProxyDAO',
+  
+  documentation: function() {/*
+    DAO Decorator that stores a fixed number of objects.
+  */},
+  
+  javaImports:[
+    'org.apache.commons.collections.buffer.CircularFifoBuffer',
+    'foam.dao.Sink',
+  ],
+  
+  properties: [
+    {
+      class: 'Object',
+      name: 'FixedSizeArray',
+      javaType: 'org.apache.commons.collections.buffer.CircularFifoBuffer',
+      javaFactory: `return new org.apache.commons.collections.buffer.CircularFifoBuffer(10000); `
```

Comment:
move the buffer size to a property, then we can set it via the builder. 
also, on buffer size change, recreate the fixedSizeArray, then we can change it at runtime via a script.